toolchain/driver/clang_runner.cpp

```diff
@@ -42,24 +50,179 @@ auto clang_main(int Argc, char** Argv, const llvm::ToolContext& ToolContext)
 namespace Carbon {
 
 ClangRunner::ClangRunner(const InstallPaths* install_paths,
-                         llvm::StringRef target,
                          llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs,
                          llvm::raw_ostream* vlog_stream)
     : ToolRunnerBase(install_paths, vlog_stream),
-      target_(target),
       fs_(std::move(fs)),
       diagnostic_ids_(new clang::DiagnosticIDs()) {}
 
+ClangRunner::~ClangRunner() {
+  if (!tmp_dir_.empty()) {
+    std::error_code ec;
+    if (std::filesystem::remove_all(tmp_dir_, ec) == 0) {
+      CARBON_VLOG("Expected a temporary directory and found none: {0}",
+                  tmp_dir_);
+    } else if (ec) {
+      CARBON_VLOG("Error when cleaning temporary directory `{0}`: {1}",
+                  tmp_dir_, ec.message());
+    }
+  }
+}
+
+// Searches an argument list to a Clang execution to determine the expected
+// target string, suitable for use with `llvm::Triple`, should be used.
```

Comment:
Small grammar nit, maybe one of:
```suggestion
// target string, suitable for use with `llvm::Triple`, that should be
// used.
```
```suggestion
// target string, suitable for use with `llvm::Triple`.
```